### ISSUE TITLE
Improve production snitches error messages when failed to open/read a configuration file(s)

### ISSUE THREAD

**Issue by vladzcloudius** [scylladb/scylla#4860]
*Installation details*
Scylla version: 3.0.9

*Description*
When snitch fails to open a configuration file the printed error message doesn't help with understanding what the problem was:

```
2019/08/16 23:25:56 ERROR 2019-08-16 23:25:56,357 [shard 0] snitch_logger - Failed to create Ec2Snitch: std::experimental::filesystem::v1::__cxx11::filesystem_error (error system:2, No such file or directory)
2019/08/16 23:25:56 ERROR 2019-08-16 23:25:56,357 [shard 0] seastar - Exiting on unhandled exception: std::experimental::filesystem::v1::__cxx11::filesystem_error (error system:2, No such file or directory)
2019/08/16 23:25:56 scylla: ./seastar/core/sharded.hh:377: seastar::sharded<T>::~sharded() [with Service = gms::feature_service]: Assertion `_instances.empty()' failed.
2019/08/16 23:25:56 Aborting on shard 0.
2019/08/16 23:25:56 Backtrace:
2019/08/16 23:25:56   0x000000000047f847
2019/08/16 23:25:56   0x00000000003994b9
2019/08/16 23:25:56   0x0000000000399779
2019/08/16 23:25:56   0x0000000000399835
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libpthread.so.0+0x000000000001288f
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libc.so.6+0x000000000003ee96
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000040800
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000030399
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000030411
2019/08/16 23:25:56   0x0000000000d37ad8
2019/08/16 23:25:56   0x0000000000300718
2019/08/16 23:25:56   /lib/x86_64-linux-gnu/libc.so.6+0x0000000000021b96
2019/08/16 23:25:56   0x0000000000373ca9
2019/08/16 23:26:42 ./run.sh: line 90: 28188 Aborted                 (core dumped) /usr/bin/scylla --log-to-syslog 0 --log-to-stdout 1 --default-log-level info --network-stack posix --num-io-queues 1 --io-properties-file /etc/scylla.d/io_properties.yaml --options-file /etc/scylla/scylla.yaml
```

We need to improve that.

**Comment by vladzcloudius** [comment]
The issue is fixed in the master and in 3.1.0-rc3.

The fixing commit is 
```
commit b19792405faee91ed9410357dc41b7cda648690f
Author: Avi Kivity <avi@scylladb.com>
Date:   Sat Apr 20 20:57:33 2019 +0300

    main: RAII-ify shutdown
    
    Instead of app-template::run_deprecated() and at_exit() hooks, use
    app_template::run() and RAII (via defer()) to stop services. This makes it
    easier to add services that do support shutdown correctly.
    
    Ref #2737
    Message-Id: <20190420175733.29454-1-avi@scylladb.com>

```

With 3.1 the error in the same situation as above looks as follows:
```
Aug 20 23:20:43 ip-172-10-254-194.ec2.internal scylla.bin[2198]:  [shard 0] init - Scylla version 3.1.0.rc3-0.20190725.bebfd7b26 starting.
Aug 20 23:20:43 ip-172-10-254-194.ec2.internal scylla.bin[2198]:  [shard 0] snitch_logger - Failed to create Ec2Snitch: std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed: No such file or directory [/etc/scylla/cassandra-rackdc.properties])
Aug 20 23:20:43 ip-172-10-254-194.ec2.internal systemd[1]: scylla-server.service: main process exited, code=exited, status=1/FAILURE
Aug 20 23:20:43 ip-172-10-254-194.ec2.internal systemd[1]: Failed to start Scylla Server.
```

And  with the master (698b72b5018868df6a839d08fd24c642db97ffcd) as follows:
```
INFO  2019-08-20 19:06:00,177 [shard 0] init - Scylla version 666.development-0.20190816.62fc5c9ae starting.
INFO  2019-08-20 19:06:00,178 [shard 0] init - starting prometheus API server
INFO  2019-08-20 19:06:00,178 [shard 0] init - creating tracing
INFO  2019-08-20 19:06:00,178 [shard 0] init - creating snitch
ERROR 2019-08-20 19:06:00,178 [shard 0] snitch_logger - Failed to parse a properties file (/home/vladz/.ccm/scylla-1/node1/conf/cassandra-rackdc.properties). Halting...
ERROR 2019-08-20 19:06:00,178 [shard 0] snitch_logger - Failed to create GossipingPropertyFileSnitch: std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed [/home/vladz/.ccm/scylla-1/node1/conf/cassandra-rackdc.properties])
INFO  2019-08-20 19:06:00,178 [shard 0] init - stopping prometheus API server
INFO  2019-08-20 19:06:00,178 [shard 0] init - Startup failed: std::filesystem::__cxx11::filesystem_error (error system:2, filesystem error: open failed [/home/vladz/.ccm/scylla-1/node1/conf/cassandra-rackdc.properties])
```

**Comment by vladzcloudius** [comment]
@tzach IMO we may safely close this issue.
Comments?
